internal/mode/static/state/graph/httproute_test.go

```diff
@@ -790,18 +791,22 @@ func TestBindRouteToListeners(t *testing.T) {
 		},
 	}
 
-	invalidNotAttachableListener := createModifiedListener("", func(l *Listener) {
-		l.Valid = false
-		l.Attachable = false
-	})
-	nonMatchingHostnameListener := createModifiedListener("", func(l *Listener) {
-		l.Source.Hostname = helpers.GetPointer[gatewayv1.Hostname]("bar.example.com")
-	})
+	invalidNotAttachableListener := func(name string) *Listener {
```

Comment:
I wonder what was the reason for introducing functions, rather than keeping values. For example:
```go
	invalidNotAttachableListener := createModifiedListener("listener-80-1", func(l *Listener) {
		l.Valid = false
		l.Attachable = false
	})
	nonMatchingHostnameListener := createModifiedListener("listener-80-1", func(l *Listener) {
		l.Source.Hostname = helpers.GetPointer[gatewayv1.Hostname]("bar.example.com")
	})
```

If we need to introduce functions,  it'd make it more consistent with the existing code if we add  `create` prefix (note the similar functions -- createListener and createModifiedListener )

Reply:
> I wonder what was the reason for introducing functions, rather than keeping values.

Can I hard-code the value of the name to `listener-80-1`? If so, keeping values makes sense.

Reply:
I updated the PR to keep values https://github.com/nginxinc/nginx-gateway-fabric/pull/1324/commits/9e9ecc64792e36bfbc9bfaac77baf1f66755061c. Let me know if hard-code is not allowed. Thanks!